enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+of a Bundle resource. The management of a Bundle and BundleDeployment resources is delegated to individual "provisioner" implementations.
+
+Both of these APIs are cluster-scoped resources.
+
+### Provisioner
+
+A "provisioner" is a controller that understands the Bundle and BundleDeployment APIs. Individual provisioner implementations use ingress-style "class" name identifiers to distinguish between the BundleDeployments and Bundles resources that it needs to manage. Each provisioner implementation requires the definition of a unique provisioner IDs.
+This provisioner pattern helps abstract the variations of bundle formats that are defined throughout the Kubernetes landscape. Future provisioner implementations may manage helm charts, crossplane packages, and OLM bundle formats.
+
+Both the Bundle and BundleDeployment APIs require a unique provisioner ID to be specified. This enables higher level systems (e.g. a PO manager) to generate BundleDeployments that referencing different unique provisioner IDs. This is useful for components that want more control over how to source, and apply, Bundle contents.
+
+## Proposal
+
+### Introducing a "PlatformOperator" CRD
+
+This first phase will introduce a new, top-level "PlatformOperator" cluster-scoped CRD. While admins today could include OLM manifests before cluster installation, this approach lacks the ability to influence the cluster's lifecycle.
+This new CRD will bring CVO awareness of these manifests, and help pave the way for adding a first-class UX (e.g. via the installer's install-config API) for configuring POs in future phases.
+
+Given this is an openshift-only API, this proposal recommends introducing the CRD into the openshift/api repository under the `platform.openshift.io/v1alpha1` API group. During phase 0, this resource will only contain a single, required `spec.packageName` field. This field will be used to configure the desired OLM-based Operator package name.
+Future phase requirements may necessitate more granular admin controls over how upgrades are approved, and how PO packages are sourced from OLM catalog content.
+
+Before cluster creation, an admin would be responsible for creating, and editing, the cluster's Kubernetes manifests with their desired PlatformOperator(s) configuration, and a FeatureGate manifest that enables the "TechPreviewNoUpgrades" feature set. An example for how to configure an individual PO resource, and how to enable POs through this FeatureGate manifest will be detailed below.
+The installer would then bootstrap the cluster using those custom manifests.
+
+### The PO Manager (POM)
+
+A PO manager (POM) component will also be introduced, which is responsible for watching and reconciling the PlatformOperator custom resources, and ensuring the set of admin desired OLM-based operators get installed on the cluster. This component will also be responsible for proxying individual PO state back to the CVO.
```

Comment:
What's the story about security/rbac and privilege escalation here?

Will this component need to be particularly highly privileged? If it's creating these bundles for another operator to consume, I guess the other component will need to be highly privileged to install rbac for the operators? Can an unprivileged user install an operator with higher privileges than themselves?

Reply:
> Will this component need to be particularly highly privileged? If it's creating these bundles for another operator to consume, I guess the other component will need to be highly privileged to install rbac for the operators?

It's the latter that will be highly privileged, and this is also an issue with current OLM. In theory, you could escalate permissions through the BundleDeployment API, which is why the drawbacks section had that behavior outlined. You'd be able limit this scenario through upfront RBAC (although it sounds like this might not be necessary reading your other comment?) to users that require these permissions.

During phase 0, the only immediate consumer of the BundleDeployment API will be the POM component. It's still possible that users could configure their own BundleDeployment resources assuming they have those permissions. In the future, we're looking at expressing approval knobs as a first-class citizen on OLM's 1.x roadmap.

Reply:
Added this as an answered question in the latest commit.